tests/01-internal/03-response.c

```diff
@@ -155,6 +155,33 @@ void t04_cookies(){
 	FAIL_IF_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key4=value4; expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/; domain=*.example.org; HttpOnly; Secure");
 	FAIL_IF_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key4=value4; domain=*.example.org; HttpOnly; path=/; Secure");
 
+	onion_dict_remove(h, "Set-Cookie");
+	onion_response_add_cookie(res, "key5", "value5", -1, "/", "*.example.org", OC_HTTP_ONLY|OC_SECURE|OC_SAMESITE_STRICT);
+	FAIL_IF_NOT_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key5=value5; path=/; domain=*.example.org; HttpOnly; Secure; SameSite=Strict");
+
+	onion_dict_remove(h, "Set-Cookie");
+	onion_response_add_cookie(res, "key6", "value6", -1, NULL, NULL, OC_SAMESITE_STRICT | OC_SAMESITE_LAX);
+	FAIL_IF_NOT_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key6=value6");
+	onion_dict_remove(h, "Set-Cookie");
+	onion_response_add_cookie(res, "key6", "value6", -1, NULL, NULL, OC_SAMESITE_STRICT | OC_SAMESITE_NONE);
+	FAIL_IF_NOT_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key6=value6");
+	onion_dict_remove(h, "Set-Cookie");
+	onion_response_add_cookie(res, "key6", "value6", -1, NULL, NULL, OC_SAMESITE_NONE | OC_SAMESITE_LAX);
+	FAIL_IF_NOT_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key6=value6");
+	onion_dict_remove(h, "Set-Cookie");
+	onion_response_add_cookie(res, "key6", "value6", -1, NULL, NULL, OC_SAMESITE_STRICT | OC_SAMESITE_LAX | OC_SAMESITE_NONE);
+	FAIL_IF_NOT_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key6=value6");
+
+	onion_dict_remove(h, "Set-Cookie");
+	onion_response_add_cookie(res, "key7", "value7", -1, NULL, NULL, OC_SAMESITE_STRICT);
+	FAIL_IF_NOT_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key7=value7; SameSite=Strict");
+	onion_dict_remove(h, "Set-Cookie");
+	onion_response_add_cookie(res, "key7", "value7", -1, NULL, NULL, OC_SAMESITE_LAX);
+	FAIL_IF_NOT_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key7=value7; SameSite=Lax");
+	onion_dict_remove(h, "Set-Cookie");
+	onion_response_add_cookie(res, "key7", "value7", -1, NULL, NULL, OC_SAMESITE_NONE);
+	FAIL_IF_NOT_EQUAL_STR(onion_dict_get(h, "Set-Cookie"), "key7=value7; SameSite=None");
+
```

Comment:
I miss in this tests also setting OC_HTTP_ONLY and OC_SECURE OR'ed with the OC_SAMESITE_*. 

Also the tests as of this version fail:

```
❯ ./03-response
03-response.c:39 INFO Test t01_create_add_free
[B63F1780] [2021-01-07 18:45:37] [ERROR response.c:270] Bad formed response. Need a request at creation. Will not write headers.
03-response.c:53 INFO t01_create_add_free ends: 3 succeses / 0 failures

03-response.c:59 INFO Test t02_full_cycle_http10
[B63F1780] [2021-01-07 18:45:37] [INFO response.c:186] [(null)] "GET (null)" 200 30 (Close connection)
03-response.c:92 INFO t02_full_cycle_http10 ends: 8 succeses / 0 failures

03-response.c:96 INFO Test t03_full_cycle_http11
[B63F1780] [2021-01-07 18:45:37] [INFO response.c:186] [(null)] "GET /" 200 30 (Keep-Alive)
03-response.c:133 INFO t03_full_cycle_http11 ends: 8 succeses / 0 failures

03-response.c:137 INFO Test t04_cookies
03-response.c:160 ERROR FAIL IF NOT EQUAL STR onion_dict_get(h, "Set-Cookie") != "key5=value5; path=/; domain=*.example.org; HttpOnly; Secure; SameSite=Strict" ("key5=value5; path=/; domain=*.example.org; HttpOnly; Secure" != "key5=value5; path=/; domain=*.example.org; HttpOnly; Secure; SameSite=Strict")
03-response.c:164 ERROR FAIL IF NOT EQUAL STR onion_dict_get(h, "Set-Cookie") != "key6=value6" ("key6=value6; HttpOnly; Secure" != "key6=value6")
03-response.c:167 ERROR FAIL IF NOT EQUAL STR onion_dict_get(h, "Set-Cookie") != "key6=value6" ("key6=value6; Secure; SameSite=Strict" != "key6=value6")
03-response.c:170 ERROR FAIL IF NOT EQUAL STR onion_dict_get(h, "Set-Cookie") != "key6=value6" ("key6=value6; HttpOnly; SameSite=Lax" != "key6=value6")
03-response.c:173 ERROR FAIL IF NOT EQUAL STR onion_dict_get(h, "Set-Cookie") != "key6=value6" ("key6=value6; HttpOnly; Secure" != "key6=value6")
03-response.c:177 ERROR FAIL IF NOT EQUAL STR onion_dict_get(h, "Set-Cookie") != "key7=value7; SameSite=Strict" ("key7=value7; Secure; SameSite=Strict" != "key7=value7; SameSite=Strict")
03-response.c:180 ERROR FAIL IF NOT EQUAL STR onion_dict_get(h, "Set-Cookie") != "key7=value7; SameSite=Lax" ("key7=value7; HttpOnly; SameSite=Lax" != "key7=value7; SameSite=Lax")
03-response.c:198 ERROR FAIL IF NOT valid_expires
[B63F1780] [2021-01-07 18:45:37] [ERROR response.c:270] Bad formed response. Need a request at creation. Will not write headers.
03-response.c:202 INFO t04_cookies ends: 6 succeses / 8 failures

03-response.c:206 INFO Test t05_printf
[B63F1780] [2021-01-07 18:45:37] [INFO response.c:186] [(null)] "GET (null)" 200 21 (Close connection)
03-response.c:228 INFO t05_printf ends: 1 succeses / 0 failures

03-response.c:232 INFO Test t06_empty
[B63F1780] [2021-01-07 18:45:37] [INFO response.c:186] [(null)] "GET (null)" 200 0 (Close connection)
03-response.c:258 INFO t06_empty ends: 2 succeses / 0 failures

03-response.c:262 INFO Test t07_large_printf
[B63F1780] [2021-01-07 18:45:37] [INFO response.c:186] [(null)] "GET (null)" 200 99999 (Close connection)
03-response.c:291 INFO t07_large_printf ends: 2 succeses / 0 failures

03-response.c:305 INFO TOTAL: 30 succeses / 8 failures


```